### PR TITLE
Regenerate manifests using the latest controller-tools release

### PR DESCRIPTION
**Pull request by hjiawei**
## Description

This change regenerates Calico manifests using the latest controller-gen utility. The newer release fixes[ an issue](https://github.com/kubernetes-sigs/controller-tools/pull/1000) where groupName is optional from k8s 1.31. This optional field breaks our outdated controller-gen.

## Related issues/PRs

Requires https://github.com/projectcalico/go-build/pull/615.

## Todos

- [ ] Tests
- [ ] Documentation
- [ ] Release note

## Release Note

```release-note
TBD
```

## Reminder for the reviewer

Make sure that this PR has the correct labels and milestone set.

Every PR needs one `docs-*` label.

- `docs-pr-required`: This change requires a change to the documentation that has not been completed yet.
- `docs-completed`: This change has all necessary documentation completed.
- `docs-not-required`: This change has no user-facing impact and requires no docs.

Every PR needs one `release-note-*` label.

- `release-note-required`: This PR has user-facing changes. Most PRs should have this label.
- `release-note-not-required`: This PR has no user-facing changes.

Other optional labels:

- `cherry-pick-candidate`: This PR should be cherry-picked to an earlier release. For bug fixes only.
- `needs-operator-pr`: This PR is related to install and requires a corresponding change to the operator.

### CODE REVIEW

File: lib.Makefile [hunk 1]
```diff
@@ -135,7 +135,7 @@ endif
 # For building, we use the go-build image for the *host* architecture, even if the target is different
 # the one for the host should contain all the necessary cross-compilation tools
 # we do not need to use the arch since go-build:v0.15 now is multi-arch manifest
-GO_BUILD_IMAGE ?= calico/go-build
+GO_BUILD_IMAGE ?= jha48/go-build
```
Comment (hjiawei) [inline lib.Makefile:138]: - [ ] I will update this once the calico/go-build changes are merged.

File: libcalico-go/Makefile [hunk 2]
```diff
@@ -52,12 +52,12 @@ gen-files: gen-crds
 ## Force a rebuild of custom resource definition yamls
 gen-crds:
 	rm -rf config/crd
-	$(DOCKER_GO_BUILD) sh -c '$(GIT_CONFIG_SSH) controller-gen  crd:crdVersions=v1 paths=./lib/apis/... output:crd:dir=config/crd/'
+	$(DOCKER_GO_BUILD) sh -c '$(GIT_CONFIG_SSH) controller-gen  crd:allowDangerousTypes=true,crdVersions=v1,deprecatedV1beta1CompatibilityPreserveUnknownFields=false paths=./lib/apis/... output:crd:dir=config/crd/'
```
Comment (hjiawei) [inline libcalico-go/Makefile:55]: *  `allowDangerousTypes` allows the float type.
*  `deprecatedV1beta1CompatibilityPreserveUnknownFields` adds the `preserveUnknownFields` field in generated files.

File: libcalico-go/lib/apis/crd.projectcalico.org/v1/clusterinformation_types.go [hunk 3]
```diff
@@ -24,7 +24,7 @@ import (
 
 // ClusterInformation contains the cluster specific information.
 // +k8s:openapi-gen=true
-// +kubebuilder:resource:scope=Cluster
+// +kubebuilder:resource:scope=Cluster,path=clusterinformations
```
Comment (hjiawei) [inline libcalico-go/lib/apis/crd.projectcalico.org/v1/clusterinformation_types.go:27]: The `path` option overwrites the built-in pluralization rules (no `s` after `information`).

File: libcalico-go/config/crd/crd.projectcalico.org_bgpfilters.yaml [hunk 4]
```diff
@@ -12,21 +11,27 @@ spec:
     listKind: BGPFilterList
     plural: bgpfilters
     singular: bgpfilter
+  preserveUnknownFields: false
```
Comment (hjiawei) [inline libcalico-go/config/crd/crd.projectcalico.org_bgpfilters.yaml:14]: @caseydavenport Do we still need this `preserveUnknownFields` flag? Is this flag set to `false` for some historical reasons, like converting CRDs from `apiextensions.k8s.io/v1beta1` to `apiextensions.k8s.io/v1`?

**Review by caseydavenport — APPROVED** [review]

File: libcalico-go/config/crd/crd.projectcalico.org_bgpfilters.yaml (hunk 4 above)
Comment (caseydavenport) [reply libcalico-go/config/crd/crd.projectcalico.org_bgpfilters.yaml:14]: To be honest... this knowledge may be lost to time. let me see if I can dig anything up, but I believe it is some arcane historical reason that we had this.

**Pull request merged**